src/jit/lowerxarch.cpp

```diff
@@ -2300,58 +2300,170 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
+
+    // We shouldn't have called in here if containingNode doesn't support containment
+    assert((Compiler::flagsOfHWIntrinsic(containingIntrinsicID) & HW_Flag_NoContainment) == 0);
+
+    bool supportsAlignedSIMDLoads   = false;
+    bool supportsGeneralLoads       = false;
+    bool supportsSIMDScalarLoads    = false;
+    bool supportsUnalignedSIMDLoads = false;
+
+    switch (category)
     {
-        // non-HWIntrinsic nodes are assumed to be unaligned loads, which are only
-        // supported by the VEX encoding.
-        return comp->canUseVexEncoding() && IsContainableMemoryOp(node);
+        case HW_Category_SimpleSIMD:
+        {
+            assert(supportsSIMDScalarLoads == false);
+
+            supportsAlignedSIMDLoads   = !comp->canUseVexEncoding() && (containingIntrinsicID != NI_SSE2_ConvertToVector128Double);
+            supportsUnalignedSIMDLoads = !supportsAlignedSIMDLoads;
+            supportsGeneralLoads       = supportsUnalignedSIMDLoads;
+
+            break;
+        }
+
+        case HW_Category_IMM:
+        {
+            switch (containingIntrinsicID)
+            {
+                case NI_SSE_Shuffle:
+                case NI_SSE2_CompareLessThan:
+                case NI_SSE2_ShiftLeftLogical:
+                case NI_SSE2_ShiftRightArithmetic:
+                case NI_SSE2_ShiftRightLogical:
+                case NI_SSE2_Shuffle:
+                case NI_SSE2_ShuffleHigh:
+                case NI_SSE2_ShuffleLow:
+                case NI_SSSE3_AlignRight:
+                case NI_SSE41_Blend:
+                case NI_SSE41_DotProduct:
+                case NI_SSE41_MultipleSumAbsoluteDifferences:
+                case NI_AVX_Blend:
+                case NI_AVX_Compare:
+                case NI_AVX_DotProduct:
+                case NI_AVX_InsertVector128:
+                case NI_AVX_Permute:
+                case NI_AVX_Permute2x128:
+                case NI_AVX2_InsertVector128:
+                case NI_AVX2_Permute2x128:
+                case NI_AVX2_ShiftLeftLogical:
+                case NI_AVX2_ShiftRightArithmetic:
+                case NI_AVX2_ShiftRightLogical:
+                {
+                    assert(supportsSIMDScalarLoads == false);
+
+                    supportsAlignedSIMDLoads   = !comp->canUseVexEncoding();
+                    supportsUnalignedSIMDLoads = !supportsAlignedSIMDLoads;
+                    supportsGeneralLoads       = supportsUnalignedSIMDLoads;
+
+                    break;
+                }
+
+                case NI_SSE2_Insert:
+                case NI_SSE41_Insert:
+                case NI_AVX_CompareScalar:
+                {
+                    assert(supportsAlignedSIMDLoads == false);
+                    assert(supportsUnalignedSIMDLoads == false);
+
+                    supportsSIMDScalarLoads = true;
+                    supportsGeneralLoads    = supportsSIMDScalarLoads;
```

Comment:
This is a matter of style, but I think it would be clearer just to set this to `true`.

Reply:
I had done it this way to more clearly indicate that it is only true because `supportsSIMDScalarLoads` is true.